### PR TITLE
OTT-1194: Fetch VAST unwrapper enable flag from DB in the vast unwrap module

### PR DESCRIPTION
**Pull request by Pubmatic-Supriya-Patil**
# Description

Please add change description or link to ticket, docs, etc.

# Checklist:

- [ ] PR commit list is unique (rebase/pull with the origin branch to keep master clean).
- [ ] JIRA number is added in the PR title and the commit message.
- [ ] Updated the `header-bidding` repo with appropiate commit id.
- [ ] Documented the new changes.

For Prebid upgrade, refer: https://inside.pubmatic.com:8443/confluence/display/Products/Prebid-server+upgrade

### CODE REVIEW

File: modules/pubmatic/openwrap/openwrap_sshb.go [hunk 1]
```diff
@@ -36,3 +39,21 @@ func (ow *OpenWrap) SetCache(c cache.Cache) {
 func (ow *OpenWrap) SetMetricEngine(m metrics.MetricsEngine) {
 	ow.metricEngine = m
 }
+
+// GetVastUnwrapEnable
```
Comment (pm-jaydeep-mohite) [inline modules/pubmatic/openwrap/openwrap_sshb.go:43]: Can we add function description ?

File: modules/pubmatic/vastunwrap/entryhook.go [hunk 2]
```diff
@@ -18,26 +20,49 @@ func getVastUnwrapperEnable(ctx context.Context, field string) bool {
...
+	queryParams := payload.Request.URL.Query()
+	source := queryParams.Get("source")
+	if queryParams.Get("sshb") == "1" {
+		vastRequestContext = models.RequestCtx{
+			VastUnwrapEnabled: getVastUnwrapperEnable(payload.Request.Context(), VastUnwrapEnabled),
+			Redirect:          true,
+		}
+	} else {
+		endpoint := openwrap.GetEndpoint(payload.Request.URL.Path, source)
+		requestExtWrapper, err := openwrap.GetRequestWrapper(payload, result)
```
Comment (pm-jaydeep-mohite) [inline modules/pubmatic/vastunwrap/entryhook.go:45]: Do we need this validation and profile validation here ?

Can't we do it in before reaching this point ?

**Review by pm-jaydeep-mohite — COMMENTED** [review]
Added one minor query, 
other changes LGTM

**Review by pm-jaydeep-mohite — APPROVED** [review]
LGTM

File: modules/pubmatic/openwrap/openwrap_sshb.go [hunk 3]
```diff
@@ -36,3 +39,19 @@ func (ow *OpenWrap) SetCache(c cache.Cache) {
 func (ow *OpenWrap) SetMetricEngine(m metrics.MetricsEngine) {
 	ow.metricEngine = m
 }
+
+// GetVastUnwrapEnabled function return vastunwrap flag from the database
+func GetVastUnwrapEnabled(rctx vastmodels.RequestCtx) bool {
```
Comment (pm-nilesh-chate) [inline modules/pubmatic/openwrap/openwrap_sshb.go:44]: Add UT

File: modules/pubmatic/openwrap/openwrap_sshb.go [hunk 4]
```diff
@@ -36,3 +39,19 @@ func (ow *OpenWrap) SetCache(c cache.Cache) {
...
+		PubID:     rctx.PubID,
+		ProfileID: rctx.ProfileID,
+		DisplayID: rctx.DisplayID,
+	}
+	partnerConfigMap, err := ow.getProfileData(rCtx, openrtb2.BidRequest{})
+	if err != nil || len(partnerConfigMap) == 0 {
+		return false
+	}
+	rCtx.PartnerConfigMap = partnerConfigMap
+	return models.GetVersionLevelPropertyFromPartnerConfig(rCtx.PartnerConfigMap, models.VastUnwrapperEnableKey) == "1"
```
Comment (pm-nikhil-vaidya) [inline modules/pubmatic/openwrap/openwrap_sshb.go:56]: can we declare "1" as constant, future changes will have minimum impact

Comment (Pubmatic-Supriya-Patil) [reply modules/pubmatic/openwrap/openwrap_sshb.go]: Updated

**Review by pm-nilesh-chate — APPROVED** [review]

**Review by pm-nikhil-vaidya — APPROVED** [review]
LGTM

**Pull request merged**